packages/cloudflare/src/cli/commands/setup-cli.ts

```diff
@@ -0,0 +1,112 @@
+import { existsSync } from "node:fs";
+import { createRequire } from "node:module";
+import path from "node:path";
+
+import { normalizeOptions } from "@opennextjs/aws/build/helper.js";
+import { printHeader, showWarningOnWindows } from "@opennextjs/aws/build/utils.js";
+import logger from "@opennextjs/aws/logger.js";
+import { unstable_readConfig } from "wrangler";
+import type yargs from "yargs";
+
+import { OpenNextConfig } from "../../api/config.js";
+import { ensureCloudflareConfig } from "../build/utils/index.js";
+
+export type WithWranglerArgs<T = unknown> = T & {
+	wranglerArgs: string[];
+	config: string | undefined;
+	env: string | undefined;
+};
+
+const nextAppDir = process.cwd();
+
+/**
+ * Setup the CLI, print necessary messages, and retrieve various options and configs.
```

Comment:
`setupCompiledAppCLI` has the same comment.

What's the difference, when to use which ?

Reply:
IIUC the callback is to do some work in the middle of this function.

Another way could be do move this to the caller:

```ts
ShowHeader();
CompileConfig(); // for build
LoadConfig(); // for other commands
ExecuteCommand();
```
Not sure if this would make the code easier to follow?

Reply:
I've changed it to have a single setupCLI function that takes an option for whether the config should be compiled or not instead. That should make it clearer what's going on.